lms/djangoapps/course_api/serializers.py

```diff
@@ -6,13 +6,16 @@
 import urllib
 
 from django.urls import reverse
+from django.contrib.auth import get_user_model
 from edx_django_utils import monitoring as monitoring_utils
 from rest_framework import serializers
 
 from openedx.core.djangoapps.content.course_overviews.models import \
     CourseOverview  # lint-amnesty, pylint: disable=unused-import
 from openedx.core.djangoapps.models.course_details import CourseDetails
 from openedx.core.lib.api.fields import AbsoluteURLField
+from common.djangoapps.student.models import CourseEnrollment, CourseAccessRole
+from openedx.core.djangoapps.user_api.accounts.serializers import AccountLegacyProfileSerializer
```

Comment:
This mentions "legacy" serializer, isn't there a more up-to-date one?

Reply:
I didn't find any other profile serializer other than `AccountLegacyProfileSerializer`. And this seems to be used while updating profile [here](https://github.com/edx/edx-platform/blob/9ecbe15d9123690319409fe04e5579e8a53e622d/openedx/core/djangoapps/user_api/accounts/api.py#L140).